services/src/main/java/io/scalecube/services/RemoteServiceInstance.java

```diff
@@ -37,6 +37,8 @@ public RemoteServiceInstance(ICluster cluster, ServiceReference serviceReference
     this.cluster = cluster;
     this.address = serviceReference.address();
     this.memberId = serviceReference.memberId();
+    this.futures = new ConcurrentHashMap<>();
+    this.cluster.listen().subscribe(message -> handleReply(message));
```

Comment:
how this can be done?
the RemoteServiceInstance needs a cluster instance and then subscribe on it
can you please elaborate?